OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -88,7 +88,7 @@ public virtual void Tick(Actor self)
 		public bool FaceTarget(Actor self, Target target)
 		{
 			DesiredFacing = Util.GetFacing(target.CenterPosition - self.CenterPosition, TurretFacing);
-			return TurretFacing == DesiredFacing;
+			return Util.TickFacing(TurretFacing, DesiredFacing.Value, info.ROT) == DesiredFacing;
```

Comment:
Are you sure? I'll write a unit test to see if it works or not.

Is there no order of execution for activities vs traits (i.e., Activities will always execute before traits or vice-versa)? Checking to see if the turret has turned or not this tick adds a lot of messy plumbing.

On 2015-04-12, at 4:39 PM, Paul Chote <notifications@github.com> wrote:

> In OpenRA.Mods.Common/Traits/Turreted.cs:
> 
> > @@ -88,7 +88,7 @@ public virtual void Tick(Actor self)
> >  		public bool FaceTarget(Actor self, Target target)
> >  		{
> >  			DesiredFacing = Util.GetFacing(target.CenterPosition - self.CenterPosition, TurretFacing);
> > -			return TurretFacing == DesiredFacing;
> > +			return Util.TickFacing(TurretFacing, DesiredFacing.Value, info.ROT) == DesiredFacing;
> This will probably end up doubling the ROT unless you remove a TickFacing from somewhere else.
> 
> ―
> Reply to this email directly or view it on GitHub.
>